docs/cli/ilab-config-current-state.md

```diff
@@ -0,0 +1,90 @@
+
+- **Groups & Commands**: Click exposes functionality for 'commands' and 'command groups'. For example, `ilab`, `data`, and `model` are all command groups, whereas `generate`, `train`, and `evaluate` are all commands. We will refer to the latter as subcommands, to clarify that they lie below `ilab` in the command hierarchy.
+
+- **Function Decorators**: CLI commands and subcommands can define their command line arguments in function decorators. Among other things, argument defaults can be defined here. For example:
+
+```python
+@click.option(
+    "--num-instructions",
+    type=click.INT,
+    default=-1,
+    hidden=True,
+)
+```
+
+- **Context**: Click provides a context (`ctx`) object that can be passed across command groups and their members. We use it mainly to pass around the configuration we generate from reading the defaults defined in code and/or the config file (as previously described). Specifically, we assign the configuration in various areas under `ctx.obj`. We elaborate on this later in the doc
+
+- **Click Extensions** `clickext.py` defines some helpful extensions to click functionality that we use throughout the config process. Specifically, `clickext` manages the funnelling of `config` defaults into `click` options. The `clickext.ConfigOptions` class maps values stored in `ctx.obj.config` and `ctx.obj.default_map` to the click arguments defined in a subcommand decorator, for example:
+
+```python
+@click.option(
+    "--data-path",
+    ...
+    cls=clickext.ConfigOption,
+    typ
+)
+```
+
+This allows the function to read the values stored in the context object to fill that variable.
+
+### Additional Args
+
+A subset of the `ilab model train` config variables were deemed more advanced and less likely to be tweaked, so they are exported to a separate section of the config names "additional_args". For example, on linux, the training additional args are stored in `~/.local/share/instructlab/internal/train_configuration/additional/additional_args.yaml`.
+
+For these, click needs separate functionality to be able to map any user-provided args to the config section. We enable this by specifying `config_sections` in the respective decorators.
+
+```python
+@click.option(
+    "--learning-rate",
+    ...
+    cls=clickext.ConfigOption,
+    config_sections="additional_arguments",
+    ...
+```
+
+### Global Config
+
+By default, `ilab` looks for and saves the config in `~/Library/Application Support/instructlab/config.yaml` on mac and `~/.config/instructlab/config.yaml` on Linux. Optionally, users can write their own config.yaml and set it's path to the `ILAB_GLOBAL_CONFIG` environment variable. Upon running `ilab config init`, the file located at that path will be copied and written to the default config path expected by `ilab`.
+
+## Config Flow for `ilab` Commands
```

Comment:
IMO this section should probably also mention the `click.core.ParameterSource` check that determines what the source of a particular field value is, since this affects what the eventual config obj in memory ends up looking like when the user runs `ilab config init` 
https://github.com/instructlab/instructlab/blob/main/src/instructlab/config/init.py#L145